README.md

```diff
@@ -18,3 +18,55 @@ pnpm dev
 ```
 
 This starts your app in development mode, rebuilding assets on file changes.
+
+## Editing and previewing the docs of TanStack projects locally
+
+The documentations for all TanStack projects except for `React Charts` are hosted on [https://tanstack.com](https://tanstack.com), powered by this Remix app.
+In production, the markdown doc pages are fetched from the GitHub repos of the projects, but in development they are read from the local file system.
+
+Follow these steps if you want to edit the doc pages of a project (in these steps we'll assume it's [`TanStack/form`](https://github.com/tanstack/form)) and preview them locally :
+
+1. Create a new directory called `tanstack`.
+
+```sh
+mkdir tanstack
+```
+
+2. Enter the directory and clone this repo and the repo of the project there.
+
+```sh
+cd tanstack
+git clone git@github.com:TanStack/tanstack.com.git
+git clone git@github.com:TanStack/form.git
+```
+
+> [!NOTE]
+> Your `tanstack` directory should look like this:
+>
+> ```
+> tanstack/
+>    |
+>    +-- form/
+>    |
+>    +-- tanstack.com/
+> ```
+
+> [!WARNING]
+> Make sure the name of the directory in your local file system matches the name of the project's repo. For example, `tanstack/form` must be cloned into `form` (this is the default) instead of `some-other-name`, because that way, the doc pages won't be found.
+
+3. Enter the `tanstack/tanstack.com` directory, install the dependencies and run the app in dev mode:
+
+```sh
+cd tanstack.com
+pnpm i
+# The app will run on https://localhost:3000 by default
+pnpm dev
+```
+
+4. Now you can visit http://localhost:3000/form/latest/docs/overview in the browser and see the changes you make in `tanstack/form/docs`.
+
+> [!NOTE]
+> The updated pages need to be manually reloaded in the browser.
+
+> [!WARNING]
+> You will need to update the `docs/config.json` file (in the project's repo) if you add a new doc page!
```

Comment:
I've also seen people getting confused about updating `<project>/docs/config.json` (they don't know when to update it). If you want, I can add a few more lines about it (based on the source code of the Remix app).

Reply:
Let's do it 😄

Reply:
So I looked into the usage of `config.json` files across the TanStack projects, but it seems like they are a bit different, so I think they should be documented in their own repos. 

Some examples of the differences:

| project | | file |
| --- |--- | --- |
| `Form` | has `menu` for core docs, `frameworkMenus` for the framework-specific docs | [config.json](https://github.com/TanStack/table/blob/89ea3efb82180b414fb19e203654d8257067150a/docs/config.json#L7) |
| `Query` | only has `menu` and framework-specific docs, no shared core docs | [config.json](https://github.com/TanStack/query/blob/e6cb6df5682d11d06e2421ae6a5281fe32f4de36/docs/config.json#L7) |
| `Table` | only has `menu` and _no_ framework-specific docs | [config.json](https://github.com/TanStack/table/blob/89ea3efb82180b414fb19e203654d8257067150a/docs/config.json#L7) |

Reply:
I think we should have a single format that supports multi framework.

Reply:
Sounds great! So far the `Form` config seems to be the one that could fit all the TanStack projects, but tomorrow I can go through all of them again to make sure this format covers everything. If it does, I'm happy to make a PR for the new, cross-project, multi-framework config format.

Reply:
Excellent!

Reply:
Ok, I checked all the projects, and it looks like the `Form` config format fits all of them if we add a `users` key to it (for TanStack Query). I opened up a PR draft with more info and a list of tasks. Can you check it, @tannerlinsley ? If you give me the green light, I'll start working on the tasks.